swarm/src/connection/pool.rs

```diff
@@ -54,10 +54,36 @@ use void::Void;
 mod concurrent_dial;
 mod task;
 
+enum ExecSwitch {
+    Executor(Box<dyn Executor + Send>),
+    LocalSpawn(FuturesUnordered<Pin<Box<dyn Future<Output = ()> + Send>>>),
+}
+
+impl ExecSwitch {
+    #[inline]
```

Comment:
Yes, I think it will have a performance benefit. Since `advance local` and `spawn` are called in the same function, the `match` statements can be reduced to one by the compiler and the inline doesn't increase binary size. `#[inline]` is more of a nudge to inline anyway.

Reply:
My intuition tells me that a match statement is not expensive and thus, unless in a hot-path, not worth optimizing away.

For the `spawn` case, given that we only spawn one task per connection and given that establishing a connection has many operations magnitude larger than a `match`, I don't think the `#[inline]` has any impact.

For `advance_local`, this triggers operations again orders of magnitude more expensive. Again I don't think the `#[inline]` has any impact.

Unless we have a benchmark that shows a performance improvement, I don't think we should be adding an `#[inline]` anywhere.

See also https://github.com/libp2p/rust-libp2p/issues/897 for past discussions on this topic.

Reply:
I see. You are right, I will remove it. I inserted it because I tried to make it a habit to inline everywhere a compiler might. I see why it's not appropriate.